octavia_f5/controller/statusmanager/status_manager.py

```diff
@@ -120,6 +119,23 @@ def heartbeat(self):
         Also updates listener_count for amphora database via update_listener_count() function. This is needed for
         scheduling decisions.
         """
+        self._metric_heartbeat.inc()
+
+        # Check availability of all devices
+        for device in self.bigips:
+            device_name = device.hostname
+            LOG.debug('Checking availability of device with URL {}'.format(device_name))
+            timeout = CONF.f5_agent.availability_timeout
+
+            # Try reaching device
+            available = True
+            try:
+                self.get(path='', timeout=timeout)
```

Comment:
I would rather like to see two instances of as3restclient

Reply:
You mean instanciating `BigipAS3RestClient` and inspecting the result of its internal call to `self.info`? I don't see how that would be anything but overhead in this case.